lib/fluent/plugin/output.rb

```diff
@@ -310,6 +310,10 @@ def configure(conf)
             log.warn "'flush_at_shutdown' is false, and buffer plugin '#{buf_type}' is not persistent buffer."
             log.warn "your configuration will lose buffered data at shutdown. please confirm your configuration again."
           end
+
+          if @flush_mode != :interval && buffer_conf.has_key?('flush_interval')
```

Comment:
1st example: No. It depends on default of plugins, and `output.rb` should warn about ignored `flush_interval` when plugins' default is not `interval`.

My first point is that we cannot know the `:interval` set in `@flush_mode` is from plugin default or configuraitons at L314 (because it can be set at L293). So we should check `conf` before `super` to show appropriate warning.

2nd example: No. IMO it's better to raise error for inconsistent configuration.

The last my comment was wrong, because `Fluent::Compat::TimeSlicedOutput#configure` will set `flush_mode` as `interval` if `conf` has `flush_interval`. So we can raise exception if both of `flush_mode lazy` and `flush_interval ...` are specified in configurations explicitly.